phare/pp/diagnostics.py

```diff
@@ -0,0 +1,234 @@
+
+    @staticmethod
+    def _extract(diag, h5item):
+        assert isinstance(diag, Diagnostic)
+        return Diagnostics._extract_recursive(diag, h5item)
+
+    @staticmethod
+    def _get_h5_files(sim: ph.Simulation):
+        assert isinstance(sim, ph.Simulation)
+
+        h5files = []
+        for root, dirs, files in os.walk(sim.diag_options["options"]["dir"]):
+            for file in files:
+                if Diagnostic.is_input_file_valid(file):
+                    h5files.append(os.path.join(root, file))
+                else:
+                    print("Warning: Input file not recognized, ignored: ", file)
+        return h5files
+
+    @staticmethod
+    def _validate_simulator(sim: ph.Simulation):
+        """Fails if the input Simulation object has no diagnostic output"""
+
+        assert isinstance(sim, ph.Simulation)
+
+        if any(
+            [
+                sim.diag_options is None,
+                sim.diag_options["options"] is None
+                or sim.diag_options["options"]["dir"] is None,
+            ]
+        ):
+            raise ValueError(
+                "Cannot load diagnostics from specific simulation: no diagnostics made"
+            )
+
+    @staticmethod
+    def _merge_particle_pops(diags):
+        """Takes per HDF5 file Diagnostic objects and coallates the particle population
+            types into one DType object."""
+
+        ndiags, particle_pops = [], {}
+        for d in diags:
+            if not isinstance(d.type, _Particle):
+                ndiags.append(d)
+            else:
+                if d.type.pop not in particle_pops:
+                    particle_pops[d.type.pop] = {}
+                particle_pops[d.type.pop][d.type.name] = d
+        for p, t in particle_pops.items():
+            ndiags.append(Particles(p, t["domain"], t["patchGhost"], t["levelGhost"]))
+        return ndiags
+
+    @staticmethod
+    def extract(sim: ph.Simulation):
+        assert isinstance(sim, ph.Simulation)
+
+        Diagnostics._validate_simulator(sim)
+        diags = [Diagnostic(sim, f) for f in Diagnostics._get_h5_files(sim)]
+        if len(diags) == 0:
+            raise ValueError(
+                "Cannot load diagnostics from specific simulation: no diagnostics found"
+            )
+        dic = {clazz.__name__: [] for clazz in [_EM, _Fluid, _Particle]}
+        [
+            dic[type(d.type).__name__].append(d)
+            for d in Diagnostics._merge_particle_pops(diags)
+        ]
+        return dic
+
+    @staticmethod
+    def get_type_from_file(file):
+        file = os.path.basename(file)
+        for Type, files in Diagnostic.h5FilesPerType.items():
+            if file in files:
+                return Type
+        for Type, regexs in Diagnostic.h5FilesRegexesPerType.items():
+            for h5File, regex in regexs:
+                if regex.match(file):
+                    return Type
+        assert False
+
+    @staticmethod
+    def get(diags, Type):
```

Comment:
`get` is not explicit enough, we don't want to look inside the function to get what we get. :)

Reply:
```
def get_diag_for_quantity(diags, quantity):
```

Reply:
`qtyDiags(diags, quantity)` ?